### PR TITLE
Fix bump.yml Worfklow

### PR DESCRIPTION
**Pull request by pbrisbin**
- Add scheduled workflow for new resolvers
- Run bump workflow when it changes
- Fixup CRLF without dos2unix

### CODE REVIEW

**Comment by pbrisbin** [comment]
There some latent lint errors, but I'd like to merge around them on this PR that just fixes the automation. The created PR for 18.23 is a better place to resolve those IMO.

File: bin/new-snapshot [hunk 1]
```diff
@@ -4,7 +4,7 @@ set -euo pipefail
 get_latest_lts() {
   curl --silent -I https://www.stackage.org/lts |
     sed '/^location: \/\(.*\)$/!d; s//\1/' |
-    dos2unix # fix up CRLF
+    tr -d '\r' # fix up CRLF
```
Comment (pbrisbin) [inline bin/new-snapshot:7]: This was the actual bug, everything else here is improvement.

File: .github/workflows/bump.yml [hunk 2]
```diff
@@ -18,8 +23,17 @@ jobs:
...
         uses: peter-evans/create-pull-request@v3
         with:
+          # Use PAT so this will trigger test/lint on the created PR
+          token: ${{ secrets.FRECKLE_AUTOMATION_GITHUB_TOKEN }}
+          base: main
           add-paths: ${{ steps.bump.outputs.path }}
           commit-message: 'Create ${{ steps.bump.outputs.to }} Snapshot'
           title: 'Create ${{ steps.bump.outputs.to }} Snapshot'
-          body: '[Resolver Changelog](https://rcl.freckle.com/?from=${{ steps.bump.outputs.from }}&to=${{ steps.bump.outputs.to }}).'
-          team-reviewers: freckle/backenders
```
Comment (pbrisbin) [inline .github/workflows/bump.yml:25]: I couldn't get this to work. So I'm adding a `CODEOWNERS` hoping that'll just trigger reviewers -- it's good to have anyway.

File: .github/workflows/bump.yml [hunk 3]
```diff
@@ -18,8 +23,17 @@ jobs:
       - if: ${{ steps.bump.outputs.bumped == 'true' }}
         uses: peter-evans/create-pull-request@v3
         with:
+          # Use PAT so this will trigger test/lint on the created PR
```
Comment (mjgpy3) [inline .github/workflows/bump.yml:26]: Huh?

File: README.md [hunk 4]
```diff
@@ -39,15 +39,28 @@ _See [this Issue][issue] for a more complete discussion._
...
+contains important packages or a GHC upgrade, etc. It may require edits to
+compile. **In general**, if our tests and linting pass, it is acceptable to
+merge.
+
+If you would like to more thorough testing, you can update any app to use the
+resolver from the branch by changing its `stack.yaml`:
+
+```diff
+- resolver: https://raw.githubusercontent.com/freckle/stackage-snapshots/main/lts/18/21.yaml
++ resolver: https://raw.githubusercontent.com/freckle/stackage-snapshots/create-pull-request/patch/lts/18/23.yaml
```
Comment (mjgpy3) [inline README.md:56]: This is good to note. Sometimes I forget this strategy.

File: README.md [hunk 5]
```diff
@@ -39,15 +39,28 @@ _See [this Issue][issue] for a more complete discussion._
...
+```
 
 ## Automated linting
 
-CI will run [LSD][], which checks for:
+CI will run [SLED][], which checks for:
 
-[lsd]: https://github.com/freckle/lsd
+[sled]: https://github.com/freckle/stack-lint-extra-deps
```
Comment (mjgpy3) [inline README.md:63]: Cool short-hand with the empty `[]`. It's case insensitive, eh?

File: bin/new-snapshot (hunk 1 above)
Comment (mjgpy3) [reply bin/new-snapshot:7]: Missing package?

**Review by mjgpy3 — APPROVED** [review]
:confetti_ball: Great!

File: bin/new-snapshot (hunk 1 above)
Comment (pbrisbin) [reply bin/new-snapshot:7]: Yup, mentioned that in the commit message for this specific change. Very surprising!

**Pull request merged**